server/src/com/thoughtworks/go/config/update/DeleteTemplateConfigCommand.java

```diff
@@ -54,7 +55,15 @@ public boolean isValid(CruiseConfig preprocessedConfig) {
 
     @Override
     public boolean canContinue(CruiseConfig cruiseConfig) {
-        return super.canContinue(cruiseConfig) && doesTemplateExist(cruiseConfig);
+        return isUserAuthorized() && doesTemplateExist(cruiseConfig);
```

Comment:
Can you see if `TemplateConfigCommand`'s `canContinue` method is correct? Should it call `isAuthorizedToEditTemplate` too, instead of `isUserAdmin`? It is being used only in `CreateTemplateConfigCommand` since you removed the only other usage of `super.canContinue` in this context.

I guess it's fine, because only pipeline group admins and above can create the template. If so, I'd make the `canContinue` in `TemplateConfigCommand` fully abstract and move the implementation down to `CreateTemplateConfigCommand`, since it's the only one using it.